app/controllers/user_products_controller.rb

```diff
@@ -62,4 +62,8 @@ def check_user_product
   def user_product
     @user_product ||= current_user.user_products.find_by(id: params[:id])
   end
+
+  def fill_products
+    @products ||= Product.order('name ASC')
```

Comment:
Memoized variable @products does not match method name fill_products. Use @fill_products instead.